# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLExtensionMOZDebug.cpp
# Commit: 71c122ac0ca7
# Full Hash: 71c122ac0ca73391866b1ef19f4f82bc2d28568b
# Author: David Parks <davidp99@gmail.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Initial out-of-process WebGL implementation. r=mccr8,handyman
#   
#   Splits WebGLContext into ClientWebGLContext and HostWebGLContext.  The Client enables the JS-control of a WebGL context in a content procecss while the Host executes the WebGL graphics operations (via a WebGLContext that maintains much of the existing code) in the compositor process.  At this point, the cross-process behavior is disabled -- this series of patches is an incremental step toward that final goal.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D54018
# ==============================================================================

diff -r 5cbf5bb11c58 -r 71c122ac0ca7 dom/canvas/WebGLExtensionMOZDebug.cpp
--- a/dom/canvas/WebGLExtensionMOZDebug.cpp	Wed Jan 08 21:52:03 2020 +0000
+++ b/dom/canvas/WebGLExtensionMOZDebug.cpp	Wed Jan 08 22:19:14 2020 +0000
@@ -17,10 +17,8 @@
 
 WebGLExtensionMOZDebug::~WebGLExtensionMOZDebug() {}
 
-void WebGLExtensionMOZDebug::GetParameter(JSContext* cx, GLenum pname,
-                                          JS::MutableHandle<JS::Value> retval,
-                                          ErrorResult& er) const {
-  if (mIsLost || !mContext) return;
+MaybeWebGLVariant WebGLExtensionMOZDebug::GetParameter(GLenum pname) const {
+  if (mIsLost || !mContext) return {};
   const WebGLContext::FuncScope funcScope(*mContext, "MOZ_debug.getParameter");
   MOZ_ASSERT(!mContext->IsContextLost());
 
@@ -43,36 +41,31 @@
           ret.Append(NS_ConvertUTF8toUTF16(rawExt));
         }
       }
-      retval.set(StringValue(cx, ret, er));
-      return;
+      return AsSomeVariant(ret);
     }
 
     case LOCAL_GL_RENDERER:
     case LOCAL_GL_VENDOR:
     case LOCAL_GL_VERSION: {
       const auto raw = (const char*)gl->fGetString(pname);
-      retval.set(StringValue(cx, NS_ConvertUTF8toUTF16(raw), er));
-      return;
+      nsString ret = NS_ConvertUTF8toUTF16(raw);
+      return AsSomeVariant(ret);
     }
 
     case dom::MOZ_debug_Binding::WSI_INFO: {
       nsCString info;
       gl->GetWSIInfo(&info);
-      retval.set(StringValue(cx, NS_ConvertUTF8toUTF16(info), er));
-      return;
+      nsString ret = NS_ConvertUTF8toUTF16(info);
+      return AsSomeVariant(ret);
     }
 
     case dom::MOZ_debug_Binding::DOES_INDEX_VALIDATION:
-      retval.set(JS::BooleanValue(mContext->mNeedsIndexValidation));
-      return;
+      return AsSomeVariant(std::move(mContext->mNeedsIndexValidation));
 
     default:
       mContext->ErrorInvalidEnumInfo("pname", pname);
-      retval.set(JS::NullValue());
-      return;
+      return Nothing();
   }
 }
 
-IMPL_WEBGL_EXTENSION_GOOP(WebGLExtensionMOZDebug, MOZ_debug)
-
 }  // namespace mozilla